### PR TITLE
Create cleaner code for barycentric coordinates

### PR DESCRIPTION
**Pull request by nhaehnle**
... and add a minor tweak to update_llpc_test_checks.py that allows checking only one or a subset of functions.

### CODE REVIEW

**Comment by nhaehnle** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit d600737fdd8b65b6d10fb6b351cbc67109254b07
<details><summary>CTS tests (Failed: 0/282025)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70507 (55.1%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31645/70507 (44.9%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70507 (55.5%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31395/70507 (44.5%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70509 (51.0%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34550/70509 (49.0%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70502 (50.9%)</li>
<li>Failed: 0/70502 (0.0%)</li>
<li>Not Supported: 34641/70502 (49.1%)</li><li>Warnings: 0/70502 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — DISMISSED** [review]
LGTM

File: lgc/builder/InOutBuilder.cpp [hunk 1]
```diff
@@ -974,47 +974,43 @@ Value *InOutBuilder::readBuiltIn(bool isOutput, BuiltInKind builtIn, InOutInfo i
...
   auto baryType = FixedVectorType::get(getFloatTy(), 3);
-  Value *barycoord = UndefValue::get(baryType);
   auto one = ConstantFP::get(getFloatTy(), 1.0);
   auto zero = ConstantFP::get(getFloatTy(), 0.0);
 
-  auto iCoord = CreateExtractElement(iJCoord, uint64_t(0));
-  auto jCoord = CreateExtractElement(iJCoord, 1);
-  auto kCoord = CreateFSub(ConstantFP::get(getFloatTy(), 1.0), iCoord);
-  kCoord = CreateFSub(kCoord, jCoord);
+  Value *hwCoord[3];
```
Comment (amdrexu) [inline lgc/builder/InOutBuilder.cpp:980]: A nit: hwCoord[3] = {};
If someone changes following codes in the future, there is still an initialized null value.

**Review by amdrexu — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit f0a7089c9019075e00b447c8d9e24d94d96d895b
<details><summary>CTS tests (Failed: 0/282025)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70507 (55.1%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31645/70507 (44.9%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70507 (55.5%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31395/70507 (44.5%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70509 (51.0%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34550/70509 (49.0%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70502 (50.9%)</li>
<li>Failed: 0/70502 (0.0%)</li>
<li>Not Supported: 34641/70502 (49.1%)</li><li>Warnings: 0/70502 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
Rebasing to trigger the CI.

**Review by amdrexu — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 57ccfbca38f6c996d09761c8cc04bbda11cfcbf3
<details><summary>CTS tests (Failed: 0/282025)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70507 (55.1%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31645/70507 (44.9%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70507 (55.5%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31395/70507 (44.5%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70509 (51.0%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34550/70509 (49.0%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70502 (50.9%)</li>
<li>Failed: 0/70502 (0.0%)</li>
<li>Not Supported: 34641/70502 (49.1%)</li><li>Warnings: 0/70502 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
Rebasing yet again, I hear CI should now work again?

**Review by amdrexu — APPROVED** [review]

**Comment by nhaehnle** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 67b565d788e1fdf8af213c90ca9e07c7b296767a
<details><summary>CTS tests (Failed: 0/282025)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38862/70507 (55.1%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31645/70507 (44.9%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70507 (55.5%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31395/70507 (44.5%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70509 (51.0%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34550/70509 (49.0%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 35861/70502 (50.9%)</li>
<li>Failed: 0/70502 (0.0%)</li>
<li>Not Supported: 34641/70502 (49.1%)</li><li>Warnings: 0/70502 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**